### ISSUE TITLE
Update script fails on /userdata/system/switch/extra/batocera-switch-tar: This does not look like a tar archive

### ISSUE THREAD

**Issue by Rcsander** [ordovice/batocera-switch#285]
Fulll error here:

/userdata/system/switch/extra/batocera-switch-tar: This does not look like a tar archive

gzip: stdin: not in gzip format
/userdata/system/switch/extra/batocera-switch-tar: Child returned status 1
/userdata/system/switch/extra/batocera-switch-tar: Error is not recoverable: exiting now
cp: cannot stat '/userdata/system/switch/extra/downloads/ryujinx/publish/lib*': No such file or directory
ls: cannot access './lib*': No such file or directory
wc: /userdata/system/switch/extra/ryujinx/Ryujinx.AppImage: No such file or directory
/tmp/batocera-switch-updater.sh: line 1239: /1048576: syntax error: operand expected (error token is "/1048576")

Has anyone seen this or worked around it? I deleted the switch system directory but still have the same issue on rebuild. Batocera v38.

**Comment by kryptk2010** [comment]
Doubt the script will work anymore, nothing to download Ryujinx from got taken down today.

**Comment by Rcsander** [comment]
Great day to update batocera...I ended up doing a fresh install, any way to relink everything so it launches the appimage and correct file from within emulation station. Maybe the wrong place to ask I know...

**Comment by kryptk2010** [comment]
No clue, i keep a backup of the entire drive, so if anything goes wrong can transfer what I need back, using a 16tb setup though it is safer to have a backup.

You mainly need the es_switch files, and system/switch folder then just place the appimages.

**Comment by Rcsander** [comment]
Yup, I can launch the appimage from terminal, and have the system/switch folder no problem. but I am struggling with the es_switch files. Seems the script no longer errors out, makes the system directory but nothing for the ES to launch and no desktop icons in the application as well...

**Comment by Rcsander** [comment]
So I was able to replace this section in the system/configs/emulationstation/es_systems_switch.cfg "/userdata/system/switch/configgen/switchlauncher.py %CONTROLLERSCONFIG% -system %SYSTEM% -rom %ROM%</command>
		

with this:

<command>/userdata/system/switch/yuzu.AppImage %ROM%</command>
		

I can launch from ES now, but wonder why the original command does not work. I'm not exactly sure what the %SYSTEM% calls...

**Comment by AshenFFXI** [comment]
https://github.com/foclabroc/batocera-switch

Looks like foclabroc fixed it in the new 41 butterflys. Ran the new script on my test system and seems to be working fine now.

**Comment by KruyKaze** [comment]
> https://github.com/foclabroc/batocera-switch
> 
> Looks like foclabroc fixed it in the new 41 butterflys. Ran the new script on my test system and seems to be working fine now.

That fixed my install. I just ran the script without deleting anything already installed. Thank you so much.

**Comment by kryptk2010** [comment]
Latest beta broke it again lol. do not update your Beta builds.

Might be related to this change 

removed #. TRANSLATION: jaguar.bigpemu.keys, dreamcast.redream.keys, switch.ryujinx.keys, vpinball.keys, theforceengine.keys, ...
added  #. TRANSLATION: steam.keys
